### PR TITLE
Remove redundant uses of virtual keyword

### LINKED ISSUE #207: STL: Finish marking virtual functions as override
We've marked most `virtual` functions as `override` when possible. We should do a final pass to find any that were missed.

Our non-Standard `_Doraise()` was one example, fixed by #302. I am unsure if there are any remaining examples.

### LINKED ISSUE #303: STL: Should we avoid repeating `virtual` for `override` functions?
Currently, we follow a convention of saying both `virtual` and `override` on derived implementations. One of many examples:

https://github.com/microsoft/STL/blob/58bb49d63d92e7a0346a05af29816aeea6b4cf0f/stl/inc/functional#L861-L863

We started doing this when `override` was new and we weren't familiar with reading it. Now that we are, should we avoid this redundancy? The advantage to doing so, aside from reducing verbosity, is that it would make the remaining `virtual` occurrences stand out. Those indicate either base declarations, or derived implementations that *should* be marked `override` (see #207 for that).

### PR DESCRIPTION
**Pull request by SuperWig**
Fixes #303. 
Fixes #207.

Opening as draft due to "decision-needed" tag, though there wasn't anyone opposing. The ones that go onto a new line do make it harder to notice the override.

### CODE REVIEW

File: stl/inc/cvt/jis_0208 [hunk 1]
```diff
@@ -41,7 +41,7 @@ namespace stdext {
             virtual ~codecvt_jis_0208() noexcept {}
```
Comment (AlexGuteniev) [inline stl/inc/cvt/jis_0208:41]: Some destructors are just `virtual`, but can be just `override`. Not sure if it needs to be changed though.

Comment (SuperWig) [reply stl/inc/cvt/jis_0208]: A couple didn't have either.

**Comment by StephanTLavavej** [comment]
We've decided that this is indeed desirable - thanks! :smile_cat:

**Comment by AlexGuteniev** [comment]
Apparently this will resolve #207 as well

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Looks great, thanks! I've pushed a merge with `main` (no conflicts, including stealth clang-format conflicts) and I found no additional occurrences in product code that need to be changed. The fact that we were missing `virtual` entirely in some locations was :scream_cat: - thank you for cleaning everything up to say `override`. :heart_eyes_cat:

**Review by CaseyCarter — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for this comprehensive cleanup! :broom: :heart_eyes_cat: :rocket:

**Pull request merged**